File: Web.ClientApp/src/shared/contracts/responses/reconnecting-init-room-data-response.ts
import {IPlayerDto} from "../player-dto";
import {IInitRoomDataResponse} from "../init-room-data-dto";
import {GameHistoryMessage} from "shared/contracts/responses/added-game-history-message-response";

export interface IReconnectingInitRoomDataResponse {
    player: IPlayerDto;
    players: Array<IPlayerDto>;
    gameHistory: Array<GameHistoryMessage>;
    initRoomData: IInitRoomDataResponse;
}

File: src/Services/App/App.Contracts/Responses/ReconnectingInitRoomDataResponse.cs
using App.Contracts.Data;
using App.Domain.Primitives;

namespace App.Contracts.Responses;

public sealed record ReconnectingInitRoomDataResponse(
    PlayerDto Player,
    InitRoomDataDto InitRoomData,
    List<GameHistoryMessage> GameHistory,
    IEnumerable<PlayerDto> Players);